### DISCUSSION TITLE
argocd CLI specification of env vars for plugins

### DISCUSSION THREAD

**Discussion by elfner** [argoproj/argo-cd#6213] [Q&A] [answered]
In the UI, when you define an app with a plugin, you can add any required env vars required by the plugin.  Is there an equivalent switch in the **argocd** CLI to specify those env vars?  If so, it's not obvious to me from the usage syntax.  Thanks.

![pluginEnvVars](https://user-images.githubusercontent.com/9096655/117841213-b719e000-b231-11eb-8eb1-f020f08d0e39.png)

**Comment by jgwest** [comment, marked as answer]
Hi @elfner , `--plugin-env` is the option you are looking for:
```
      --plugin-env stringArray                    Additional plugin envs
```
(https://argoproj.github.io/argo-cd/user-guide/commands/argocd_app_create/)

I double-checked the code to verify it does what you are looking for.

**Reply by elfner** [reply]
Thanks - that was it.  I had an older client version w/o that switch.
